### PR TITLE
Expose WCPay refund transaction id for refund endpoint

### LINKED ISSUE #7857: Expose WooPayments transaction id (txn_xxxxx) via current refund API end
### Description

The current WooComemrce REST API only returns an internal id after a refund is created.
https://woocommerce.github.io/woocommerce-rest-api-docs/#create-a-refund

It posts a challenge for reconciliation for refunds created by NetSuite via the REST API. 
The internal id in the “Create a refund" API point cannot be reconciled with the WooPayments transaction report (Payments -> Transactions).

### Acceptance criteria

The response of the current refund API should include a new field if the payment gateway is WooPayments. https://woocommerce.github.io/woocommerce-rest-api-docs/#create-a-refund

For example, a new field should be woopayments_transaction_id: "txn_xxxxxxx" should be returned by the API endpoint.

### Testing instructions

1. Create a refund via https://woocommerce.github.io/woocommerce-rest-api-docs/#create-a-refund
2. Look for the WooPayment transactions id in the API response. 
3. Reconcile with the report in "Payments -> Transactions"

### Additional context
P2: https://woocustomersuccess.wordpress.com/2023/12/06/journey-to-launch-woopayments-for-joola-usa/

### PR DESCRIPTION
**Pull request by dpaun1985**
Fixes #7857 

#### Changes proposed in this Pull Request
Expose refund transaction id for the refund API endpoint. The transaction id is saved on the refund order in the metadata, with `_refund_transaction_id ` key.

#### Testing instructions

* Create REST API Keys as described [here](https://woocommerce.github.io/woocommerce-rest-api-docs/#rest-api-keys)
* create a refund via via https://woocommerce.github.io/woocommerce-rest-api-docs/#create-a-refund . 
* check the api response, the metadata should contain `_refund_transaction_id`
* check the transactions in `Payments -> Transactions`

https://github.com/Automattic/woocommerce-payments/assets/82826872/7ec9798c-6149-4dae-92be-f9e1039abd62

-------------------

- [x] Run `npm run changelog` to add a changelog file, choose `patch` to leave it empty if the change is not significant. You can add multiple changelog files in one PR by running this command a few times. 
- [x] Covered with tests (or have a good reason not to test in description ☝️)
- [x] Tested on mobile (or does not apply)

**Post merge**

- [ ] Link to testing instructions from [release testing doc](https://github.com/Automattic/woocommerce-payments/wiki/Release-testing-instructions) following [these instructions](https://github.com/Automattic/woocommerce-payments/wiki/How-to-write-good-manual-testing-scenarios) : _Add link here / 'QA Testing Not Applicable'_
- [ ] Add or update [critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Critical-flows) and [testing instructions for critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Testing-instructions-for-critical-flows), if applicable.
- [ ] Add what's changed (description, screenshot, demo videos etc.) to the release announcement post, if applicable.

### CODE REVIEW

**Comment by botwoo** [comment]
### Test the build 

#### Option 1. Jetpack Beta

- Install and activate [Jetpack Beta](https://jetpack.com/download-jetpack-beta/).
- Use this build by searching for PR number `7932` or branch name `add/transaction-id-to-refund-endpoint` in your-test.site/wp-admin/admin.php?page=jetpack-beta&plugin=woocommerce-payments          

#### Option 2. Jurassic Ninja - available for logged-in A12s

:rocket: [Launch a JN site with this branch](https://jurassic.ninja/create/?jetpack-beta&shortlived&nojetpack&woocommerce&woocommerce-payments-dev-tools&branches.woocommerce-payments=add/transaction-id-to-refund-endpoint) :rocket:

:information_source: Install this [Tampermonkey script](https://github.com/Automattic/woocommerce-payments/tree/develop/bin/wcpay-live-branches) to get more options.

---

Build info: 

- Latest commit: afe02f963518b3a6ce90adc5e41501631c8f74b5
- Build time: 2024-01-05 15:12:10 UTC

_Note: the build is updated when a new commit is pushed to this PR._

File: changelog/add-transaction-id-to-refund-endpoint [hunk 1]
```diff
@@ -0,0 +1,4 @@
+Significance: minor
+Type: add
+
+Expose WooCommerce transaction id for a refund on refund API endpoint
```
Comment (anu-rock) [inline changelog/add-transaction-id-to-refund-endpoint]:
```suggestion
Expose the refund transaction ID in WooCommerce Order Refund API
```

File: includes/class-wc-payments-order-service.php [hunk 2]
```diff
@@ -83,6 +83,13 @@ class WC_Payments_Order_Service {
 	 */
 	const WCPAY_REFUND_ID_META_KEY = '_wcpay_refund_id';
 
+	/**
+	 * Meta key used to store WCPay refund transaction id.
+	 *
+	 * @const string
+	 */
+	const WCPAY_TRANSACTION_ID_META_KEY = '_refund_transaction_id';
```
Comment (anu-rock) [inline includes/class-wc-payments-order-service.php]: IMO, it should be more precisely named as `WCPAY_REFUND_TRANSACTION_ID_META_KEY`.

Also, I noticed that all constants that start with `WCPAY_` are linked to key names that also start with `_wcpay_`. Going by that logic, it should be:

```suggestion
	const WCPAY_REFUND_TRANSACTION_ID_META_KEY = '_wcpay_refund_transaction_id';
```

File: includes/class-wc-payments-order-service.php [hunk 3]
```diff
@@ -651,6 +658,20 @@ public function set_wcpay_refund_id_for_order( $order, $wcpay_refund_id ) {
...
 
+	/**
+	 * Set the payment metadata for refund transaction id.
+	 *
+	 * @param  mixed  $order The order.
+	 * @param  string $wcpay_transaction_id The value to be set.
+	 *
+	 * @throws Order_Not_Found_Exception
+	 */
+	public function set_wcpay_transaction_id_for_order( $order, $wcpay_transaction_id ) {
```
Comment (anu-rock) [inline includes/class-wc-payments-order-service.php]: Same comment for naming as above.

Comment (anu-rock) [reply includes/class-wc-payments-order-service.php]: Could you also add type hints for method params?

**Review by anu-rock — COMMENTED** [review]
Manual testing worked well 🎉 

___

#### Review checklist

| Parameter  | Result |
| ------------- | ------------- |
| Approach  | ✅ All good |
| Implementation  | ✅ All good |
| Automated tests  |  ✅ All good |
| Naming  | 👍  Almost good (left comments) |
| Typing  | 👍  Almost good (left a comment) |
| Code comments  | ✅ All good |
| Changelog  | 👍  Almost good (left a comment) |
| Testing instructions  | ✅ All good |
| Demo (video or JN site)  | ✅ All good |
| Manual testing  | ✅ Works as expected |

File: changelog/add-transaction-id-to-refund-endpoint (hunk 1 above)
Comment (dpaun1985) [reply changelog/add-transaction-id-to-refund-endpoint]: updated in https://github.com/Automattic/woocommerce-payments/pull/7932/commits/f05ac4211d57c00f76794c5b13fdddb6e14b9dc6 . thx

File: includes/class-wc-payments-order-service.php (hunk 2 above)
Comment (dpaun1985) [reply includes/class-wc-payments-order-service.php]: changed in https://github.com/Automattic/woocommerce-payments/pull/7932/commits/f05ac4211d57c00f76794c5b13fdddb6e14b9dc6 . thx

File: includes/class-wc-payments-order-service.php (hunk 3 above)
Comment (dpaun1985) [reply includes/class-wc-payments-order-service.php]: updated in https://github.com/Automattic/woocommerce-payments/pull/7932/commits/f05ac4211d57c00f76794c5b13fdddb6e14b9dc6 . thank you!

**Review by anu-rock — APPROVED** [review]
LGTM after the latest changes 🎉

**Review by mordeth — APPROVED** [review]
✅ Code changes are looking good!
✅ Manual tests are successful!

**Pull request merged**